compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/optimizer/BIRBasicBlockOptimizer.java

```diff
@@ -116,13 +116,18 @@ private void updatePredecessorMap(BIRBasicBlock removable) {
 
     private Set<BIRBasicBlock> getRemovableBasicBlocks(BIRNode.BIRFunction birFunction,
                                                        BIROptimizer.OptimizerEnv funcEnv) {
+        Set<BIRBasicBlock> errorTableTargetBBs = new HashSet<>();
+        for (BIRNode.BIRErrorEntry birErrorEntry : birFunction.errorTable) {
+            errorTableTargetBBs.add(birErrorEntry.targetBB);
+        }
+
         Set<BIRBasicBlock> removableBasicBlocks = new HashSet<>();
         for (int i = birFunction.basicBlocks.size() - 1; i >= 0; --i) {
             BIRBasicBlock basicBlock = birFunction.basicBlocks.get(i);
 
             // Remove basic blocks with unnecessary jump statement
             if (basicBlock.terminator instanceof BIRTerminator.GOTO && basicBlock.instructions.isEmpty()
-                    && basicBlock.terminator.pos == null) {
+                    && basicBlock.terminator.pos == null && !errorTableTargetBBs.contains(basicBlock)) {
```

Comment:
Since we are not going to remove `trap` and `end` BBs, can't we simply check whether the `errorTable` contains the basicblock which we are consider for removing?

Reply:
Do you mean there is a possibility where the previous conditions will allow the removal of `trap` and `end` BBs, and we need to also include them in the `errorTableTargetBBs` set?
If not, we can insert only the `target` BBs like this, because it will be efficient.